flang/lib/Lower/Bridge.cpp

```diff
@@ -1597,13 +1598,15 @@ class FirConverter : public Fortran::lower::AbstractConverter {
           return;
         } else
           global = builder->createGlobal(
-              toLocation(), genType(sym), globalName, false,
+              toLocation(), genType(sym), globalName, isConst,
               [&](Fortran::lower::FirOpBuilder &builder) {
                 auto initVal = genExprValue(details->init().value());
                 builder.create<fir::HasValueOp>(toLocation(), initVal);
               });
-      } else
+      } else {
+        assert (!isConst && "Missing initialization for parameter");
```

Comment:
The front end issues an error in such cases, shall I get rid of this assert?

Reply:
This `else` pairs with the test that details holds an `init`, correct?

It's perfectly legitimate for a global variable to not have an initializer, which is what the next line is creating. So looks like 2 things here.

1. You don't want to assert that a variable must be constant.
2. You don't need to assert when a constant doesn't have an initializer, since we shouldn't get here, as you say.

Reply:
(The `else` should follow the same style of the `if` part. Here, since the `if` has braces, the `else` should as well.)

Reply:
Ok I will remove the assert. I will keep the braces.

Reply:
I have removed this assert. Ready to merge.